File: code/php/sdk/tests/Unit/MergeFragmentsTest.php
<?php

use starfederation\datastar\Consts;
use starfederation\datastar\enums\FragmentMergeMode;
use starfederation\datastar\events\MergeFragments;

test('Options are correctly output', function() {
    $content = '<div>content</div>';
    $event = new MergeFragments($content, [
        'selector' => 'selector',
        'mergeMode' => FragmentMergeMode::Append,
        'settleDuration' => 1000,
        'useViewTransition' => true,
    ]);
    expect($event->getDataLines())
        ->toBe([
            'data: selector selector',
            'data: mergeMode append',
            'data: settleDuration 1000',
            'data: useViewTransition true',
            'data: fragments ' . $content,
        ]);
});

test('Default options are not output', function() {
    $content = '<div>content</div>';
    $event = new MergeFragments($content, [
        'selector' => '',
        'mergeMode' => FragmentMergeMode::Morph,
        'settleDuration' => Consts::DEFAULT_SETTLE_DURATION,
        'useViewTransition' => false,
    ]);
    expect($event->getDataLines())
        ->toBe([
            'data: fragments ' . $content,
        ]);
});

test('Multi-line content is correctly output', function() {
    $content = '<div>content</div>';
    $event = new MergeFragments("\n" . $content . "\n" . $content . "\n");
    expect($event->getDataLines())
        ->toBe([
            'data: fragments ' . $content,
            'data: fragments ' . $content,
        ]);
});

File: sdk/java/src/test/java/StarFederation/Datastar/unit/MergeFragmentsTest.java
package StarFederation.Datastar.unit;

import StarFederation.Datastar.enums.EventType;
import StarFederation.Datastar.enums.FragmentMergeMode;
import StarFederation.Datastar.events.MergeFragments;
import StarFederation.Datastar.events.MergeFragmentsOptions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;
import static StarFederation.Datastar.Consts.*;

class MergeFragmentsTest {

    private MergeFragmentsOptions defaultOptions;

    @BeforeEach
    void setUp() {
        defaultOptions = MergeFragmentsOptions.create();
    }

    @Test
    void getDataLinesShouldIncludeAllOptions() {
        String data = "<div id=\"feed\">\n<span>1</span>\n</div>";
        MergeFragmentsOptions options = MergeFragmentsOptions.create()
                .withSelector("#feed")
                .withMergeMode(FragmentMergeMode.Append)
                .withSettleDuration(10)
                .withViewTransition(true);

        MergeFragments mergeFragments = new MergeFragments(data, options);

        String[] expectedDataLines = new String[]{
                "data: " + SELECTOR_DATALINE_LITERAL + "#feed",
                "data: " + MERGE_MODE_DATALINE_LITERAL + "append",
                "data: " + SETTLE_DURATION_DATALINE_LITERAL + "10",
                "data: " + USE_VIEW_TRANSITION_DATALINE_LITERAL + "true",
                "data: " + FRAGMENTS_DATALINE_LITERAL + "<div id=\"feed\">",
                "data: " + FRAGMENTS_DATALINE_LITERAL + "<span>1</span>",
                "data: " + FRAGMENTS_DATALINE_LITERAL + "</div>"
        };

        System.out.println(Arrays.toString(mergeFragments.getDataLines()));
        assertArrayEquals(expectedDataLines, mergeFragments.getDataLines());
    }

    @Test
    void getDataLinesShouldExcludeDefaultValues() {
        String data = "<div id=\"content\">Content</div>";
        MergeFragmentsOptions options = MergeFragmentsOptions.create()
                .withMergeMode(FragmentMergeMode.Morph) // Default value
                .withSettleDuration(300)                // Default value
                .withViewTransition(false);             // Default value

        MergeFragments mergeFragments = new MergeFragments(data, options);

        String[] expectedDataLines = new String[]{
                "data: " + FRAGMENTS_DATALINE_LITERAL + "<div id=\"content\">Content</div>"
        };

        System.out.println(Arrays.toString(mergeFragments.getDataLines()));
        assertArrayEquals(expectedDataLines, mergeFragments.getDataLines());
    }

    @Test
    void getEventTypeShouldReturnCorrectType() {
        String data = "<div>sample</div>";
        MergeFragments mergeFragments = new MergeFragments(data, defaultOptions);

        assertEquals(EventType.MergeFragments, mergeFragments.getEventType());
    }

    @Test
    void constructorShouldThrowExceptionForNullData() {
        assertThrows(IllegalArgumentException.class, () -> {
            new MergeFragments(null, defaultOptions);
        });
    }

    @Test
    void constructorShouldThrowExceptionForEmptyData() {
        assertThrows(IllegalArgumentException.class, () -> {
            new MergeFragments("   ", defaultOptions);
        });
    }
}
